agent/consul/leader_connect_ca.go

```diff
@@ -407,6 +407,7 @@ func (c *CAManager) initializeRootCA(provider ca.Provider, conf *structs.CAConfi
 	if err != nil {
 		return fmt.Errorf("error getting root cert: %v", err)
 	}
+	rootPEM = ca.AddSingleNewline(rootPEM)
```

Comment:
This line isn't necessary right? Because we're parsing a single cert here and `pem.Decode` handles a missing trailing newline just fine.